File: scratch/recolor_arrow.py
import sys
from PIL import Image

def recolor_arrow():
    img_path = r'e:\Java\MinecraftMod\DarkGrey\src\main\resources\assets\dark_grey\textures\entity\itanis_arrow.png'
    img = Image.open(img_path).convert('RGBA')
    width, height = img.size
    
    # We want: main body black, small amounts of gold and white.
    # Current arrow is mostly blue/cyan.
    # Light blues -> White or Gold
    # Mid blues -> Dark Grey / Black
    # Dark blues -> Black
    
    pixels = img.load()
    for y in range(height):
        for x in range(width):
            r, g, b, a = pixels[x, y]
            if a > 0:
                # Calculate lightness
                lightness = (r + g + b) / (3.0 * 255.0)
                
                # If it's a very light pixel (like the glowing edges or tip), make it Gold or White
                if lightness > 0.8:
                    # White
                    pixels[x, y] = (255, 255, 255, a)
                elif lightness > 0.6:
                    # Gold
                    pixels[x, y] = (255, 215, 0, a)
                elif lightness > 0.4:
                    # Dark Gold
                    pixels[x, y] = (184, 134, 11, a)
                else:
                    # Black / Very dark grey for the main body
                    val = int(lightness * 255) // 2
                    pixels[x, y] = (val, val, val, a)
                    
    img.save(img_path)
    print("Recolored successfully!")

if __name__ == "__main__":
    recolor_arrow()

File: scratch/RecolorArrow.java
import javax.imageio.ImageIO;
import java.awt.image.BufferedImage;
import java.io.File;

public class RecolorArrow {
    public static void main(String[] args) throws Exception {
        File file = new File("src/main/resources/assets/dark_grey/textures/entity/itanis_arrow.png");
        BufferedImage img = ImageIO.read(file);
        
        int width = img.getWidth();
        int height = img.getHeight();
        
        for (int y = 0; y < height; y++) {
            for (int x = 0; x < width; x++) {
                int argb = img.getRGB(x, y);
                int a = (argb >> 24) & 0xff;
                int r = (argb >> 16) & 0xff;
                int g = (argb >> 8) & 0xff;
                int b = argb & 0xff;
                
                if (a > 0) {
                    double lightness = (r + g + b) / (3.0 * 255.0);
                    
                    int newR, newG, newB;
                    if (lightness > 0.8) {
                        // Brilliant White/Gold Highlight
                        newR = 255; newG = 255; newB = 220;
                    } else if (lightness > 0.6) {
                        // Pure Gold
                        newR = 255; newG = 215; newB = 0;
                    } else if (lightness > 0.4) {
                        // Goldenrod
                        newR = 218; newG = 165; newB = 32;
                    } else {
                        // Dark Gold
                        newR = 184; newG = 134; newB = 11;
                    }
                    
                    int newArgb = (a << 24) | (newR << 16) | (newG << 8) | newB;
                    img.setRGB(x, y, newArgb);
                }
            }
        }
        
        ImageIO.write(img, "png", file);
        System.out.println("Recolored successfully to pure gold!");
    }
}
